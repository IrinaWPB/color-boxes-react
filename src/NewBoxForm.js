import React, { useState } from 'react'

const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
        width: "",
        height: "",
        color: ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE)

    //Changing state every time user presses a key
    const handleChange = e => {
        const { name, value } = e.target
        setFormData(data => ({
            ...data,
            [name]: value
        }))
    }

    const handleSubmit = e => {
        e.preventDefault()
        addBox({...formData})
        setFormData(INITIAL_STATE)
    }

    return (
        <form onSubmit={handleSubmit} data-testid="form">
            <label htmlFor='width'>Enter width:</label>
            <input id="width" placeholder='width' type="number" name="width"
                   value={formData.width} onChange={handleChange} />
            <label htmlFor='height'>Enter height:</label>
            <input id="height" placeholder='height' type="number" name="height"
                   value={formData.height} onChange={handleChange} />
            <label htmlFor='color'>Enter color:</label>
            <input id="color" placeholder='color' type="text" name="color"
                   value={formData.color} onChange={handleChange} />
            <br></br>
            <button>Add a box</button>
        </form>
    )
}

export default NewBoxForm