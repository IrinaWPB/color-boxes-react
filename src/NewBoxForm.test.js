import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import NewBoxForm from './NewBoxForm'

it('renders without crashing', () => {
    render(<NewBoxForm />)
})

it('matches the snapshat', () => {
    const { asFragment } = render(<NewBoxForm />)
    expect(asFragment()).toMatchSnapshot()
})

it('should show form', () => {
    const { getByText } = render(<NewBoxForm />)
    expect(getByText('Enter width:')).toBeInTheDocument()
    expect(getByText('Enter height:')).toBeInTheDocument()
})
