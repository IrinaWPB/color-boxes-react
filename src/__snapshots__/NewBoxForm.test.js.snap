// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshat 1`] = `
<DocumentFragment>
  <form
    data-testid="form"
  >
    <label
      for="width"
    >
      Enter width:
    </label>
    <input
      id="width"
      name="width"
      placeholder="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Enter height:
    </label>
    <input
      id="height"
      name="height"
      placeholder="height"
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Enter color:
    </label>
    <input
      id="color"
      name="color"
      placeholder="color"
      type="text"
      value=""
    />
    <br />
    <button>
      Add a box
    </button>
  </form>
</DocumentFragment>
`;
